### DISCUSSION TITLE
Upload external mask in img2img

### DISCUSSION THREAD

**Discussion by A2va** [hlky/stable-diffusion-webui#306] [Feature suggestions]
For now on the mask must be draw with the brush in webui and it's not very praticable. My idea is to draw the mask in a external editor and upload it in webui along with original image.

PR merged #636

**Comment by theseus232** [comment]
Yes please. Right now the masking function isnt too useful without the ability to use a proper mask.

**Reply by matthewfcarlson** [reply]
I was wondering if I was crazy for not seeing an option to add a mask.

**Comment by alejoar** [comment]
I've seen @hlky suggest the usage of photoshop for masking a few times (e.g. https://github.com/hlky/stable-diffusion/issues/123) 

What I am assuming as masking is making the part of the image I want to mask transparent and export as png. However, img2img seems to be ignoring the transparent parts. 

I am not clear on what would be the right steps to mask in a third party editor if not making transparent.
